topk-js/src/document.rs

```diff
@@ -0,0 +1,227 @@
+use napi::{
+  bindgen_prelude::{check_status, FromNapiValue, Null, ToNapiValue},
+  sys::{self, TypedarrayType},
+};
+use napi_derive::napi;
+use std::{collections::HashMap, ptr};
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Value {
+  String(String),
+  F64(f64),
+  Bool(bool),
+  U32(u32),
+  U64(u64),
+  I32(i32),
+  I64(i64),
+  F32(f32),
+  Binary(Vec<u8>),
+  Vector(Vector),
+  Null,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Vector {
+  Float(Vec<f32>),
+  Byte(Vec<u8>),
+}
+
+impl FromNapiValue for Value {
+  unsafe fn from_napi_value(
+    env: napi::sys::napi_env,
+    napi_val: napi::sys::napi_value,
+  ) -> napi::Result<Self> {
+    let mut result: i32 = 0;
+    napi::sys::napi_typeof(env, napi_val, &mut result);
+
+    match result {
+      napi::sys::ValueType::napi_string => {
+        Ok(Value::String(String::from_napi_value(env, napi_val)?))
+      }
+      napi::sys::ValueType::napi_number => Ok(Value::F64(f64::from_napi_value(env, napi_val)?)),
+      _ => panic!("unsupported value type: {:?}", result),
+    }
+  }
+}
+
+impl ToNapiValue for Value {
+  unsafe fn to_napi_value(env: sys::napi_env, val: Self) -> Result<sys::napi_value, napi::Error> {
+    match val {
+      Value::String(s) => String::to_napi_value(env, s),
+      Value::F64(n) => f64::to_napi_value(env, n),
+      Value::Bool(b) => bool::to_napi_value(env, b),
+      Value::U32(n) => u32::to_napi_value(env, n),
+      Value::U64(n) => u32::to_napi_value(env, n as u32),
```

Comment:
this is a not a safe cast

Reply:
I replaced it with:
```rs
Value::U64(n) => i64::to_napi_value(env, n.try_into().unwrap()),
```
According to rust compiler: 
```
you can convert a `u64` to an `i64` and panic if the converted value doesn't fit: `.try_into().unwrap()`
```

Reply:
shouldn't we map to `BigInt` here?

Reply:
I would avoid using `BigInt` in JS-land since it's exported as a custom class in javascript thus unfamiliar with users. Maybe there's a use-case for that but I believe in majority of cases any number from rust should just be a `number` type in javascript